### ISSUE TITLE
`--chain` causes hanging up forever

### ISSUE THREAD

**Issue by yegor256** [objectionary/eo-phi-normalizer#548]
This is my `foo.phi` file:

```
{⟦ m ↦ ⟦ x ↦ ξ.t, φ ↦ ⟦ t ↦ ⟦⟧ ⟧ ⟧.x ⟧}
```

I run and it works just fine:

```
$ eo-phi-normalizer rewrite -s foo.phi
{
  ⟦
    m ↦ ⟦
      ρ ↦ ⟦
        t ↦ ⟦ ⟧, ρ ↦ ⟦
          x ↦ ξ.t, φ ↦ ⟦
            t ↦ ⟦ ⟧
          ⟧
        ⟧
      ⟧
    ⟧
  ⟧
}
```

Then, I go:

```
$ eo-phi-normalizer rewrite --chain foo.phi
```

It is stuck forever. If I add `--tex`, I see a long repeating console output.

**Comment by yegor256** [comment]
@deemp please, check

**Comment by deemp** [comment]
cc @fizruk

**Comment by yegor256** [comment]
@deemp this problem is solved in 2.2.1, thanks!
